File: Power of Four.cpp
class Solution {
public:
	float logn(int n, int k) {
		return log(n) / log(k) ;
	}

	bool isPowerOfFour(int num) {
		if (num == 0) return false ;
		return ceil(logn(num, 4)) == floor(logn(num, 4)) ;
	}
};

File: Power of Four.java
class Solution {
	public double logn(Integer num, Integer base) {
		return (Math.log(num)) / (Math.log(base)) ;
	}

	public boolean isPowerOfFour(int num) {
		if (num == 0) return false ;
		return Math.ceil(logn(num, 4)) == Math.floor(logn(num, 4)) ;
	}
}